packages/nextjs/utils/scaffold-eth/fetchPriceFromUniswap.ts

```diff
@@ -1,15 +1,25 @@
-import { Fetcher, Route, Token, WETH } from "@uniswap/sdk";
+import { Fetcher, Route, Token } from "@uniswap/sdk";
 import { Provider } from "@wagmi/core";
+import { getTargetNetwork } from "~~/utils/scaffold-eth";
 
 export const fetchPriceFromUniswap = async (provider: Provider): Promise<number> => {
+  const configuredNetwork = getTargetNetwork();
+  if (configuredNetwork.nativeCurrency.symbol !== "ETH" && !configuredNetwork.tokenAddress) {
+    return 0;
+  }
   try {
     const DAI = new Token(1, "0x6B175474E89094C44Da98b954EedeAC495271d0F", 18);
-    const pair = await Fetcher.fetchPairData(DAI, WETH[DAI.chainId], provider);
-    const route = new Route([pair], WETH[DAI.chainId]);
+    const TOKEN = await Fetcher.fetchTokenData(
+      1,
+      configuredNetwork.tokenAddress || "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
```

Comment:
I think this can be improved because its kind of confusing setting fallback here(which won't work and works only for one condition when on `mainnnet`) since kind of conflicting guard check above (am I missing something ? ) : 
```ts
if (configuredNetwork.nativeCurrency.symbol !== "ETH" && !configuredNetwork.tokenAddress) {
    return 0;
  }
```

Also I think for consistency, lets move WETH (0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2) address to `networks.ts`  and add : 

```ts
[chains.mainnet.id]: {
    color: "#ff8b9e",
     tokenAddress : "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
  },
```

And now, in this file we can do : 
```ts
export const fetchPriceFromUniswap = async (provider: Provider): Promise<number> => {
  const configuredNetwork = getTargetNetwork();
  if (!configuredNetwork.tokenAddress)  return 0;
  try {
      //  ...
      const TOKEN = await Fetcher.fetchTokenData(1, configuredNetwork.tokenAddress, provider);
     // ...
  }
  // ...
 }

```

Reply:
> It will work for every network that has ETH as their native currency, not only mainnet! There are a lot of them: goerli, sepolia, arbitrum, optimism, etc.

Ohhhhh, just noticed and makes sense,  lol I edited the comment at the same time 😆..... BTW then if we use the WETH address for `arbitrum`, `opitimism` etc again I think it might get misleading  

Yeah it's kindof of a PITA to add those addresses to each network :( 

BTW, what's our plan ? Do we plan to show a fallback, which returns the ETH price if it doesn't have `nativeCurrencyToken` address or we just don't allow the user to switch to USD if it's not available? 

depending on this we can use one of the both approach

Reply:
> If I think about the Balance component... my vote will be to disable the switch to USD if it's not available. If we default ro the ETH price, it can be confusing, right ?

Yes 💯, So I think maybe we keep `nativeCurrencyTokenAddress` for each token in `NETWORKS_EXTRA_DATA ` (covers famous chain and that too big) making it as source of truth and if its not there then return 0 also leading to disabling switch ot USD functionality...

Reason : 
1. Having  `configuredNetwork.nativeCurrency.symbol !== "ETH"` might again mislead on chains like `arbitrium`, `optmism` showing actual ETH value 
2. It will kind of make it consistent (I think), and will also help the developer grasp it in one go if he wants to add his own chain

But cc @damianmarti for thoughts because even I don't feel too strong about the above approach and ok with anything

Reply:
> So I think maybe we keep `nativeCurrencyTokenAddress` for each token in `NETWORKS_EXTRA_DATA ` (covers famous chain and that too big) making it as source of truth and if its not there then return 0 also leading to disabling switch ot USD functionality...

Yup that is exactly how it has been implemented currently. You might want to read more about the current flow here https://github.com/scaffold-eth/scaffold-eth-2/pull/324#discussion_r1176580525

> 1. Having  `configuredNetwork.nativeCurrency.symbol !== "ETH"` might again mislead on chains like `arbitrium`, `optmism` showing actual ETH value

Can you elaborate on this more because I am unable to understand how will it be misleading?

Reply:
So example if my targetNetwork is `arbitrum` (who's `nativeCurrency.symbol` is `ETH`) now in current situation we will first check : 
```ts

if (configuredNetwork.nativeCurrency.symbol !== "ETH" && 
//.   ^ This condition will be false and we don't go inside if 
!configuredNetwork.tokenAddress) {
    return 0;
  }
```

after that, we reach : 
```ts

// Here we will pick 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2 (WETH) as token address instead the correct address should be 0xB50721BCf8d664c30412Cfbc6cf7a15145234ad1 (ARB token address)
const TOKEN = await Fetcher.fetchTokenData(
      1,
      configuredNetwork.tokenAddress || "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      provider,
    );
```

So chains like `arbitrum` , `optimism`  etc whose symbold is `ETH` will always show `WETH / DAI` price istead they should have shown `ARB / DAI` etc which is misleading right ? Really sorry if I am missing something please feel to correct me 🙏

Reply:
> So chains like `arbitrum` , `optimism` etc whose symbold is `ETH` will always show `WETH / DAI` price istead they should have shown `ARB / DAI` etc which is misleading right ? Really sorry if I am missing something please feel to correct me 🙏

If I am not wrong the native token of `arbitrum` is `ETH` itself and not `ARB` (`ETH` is used to pay gas fees and also fetching account balance will also return the amount of `ETH` and not `ARB` in the wallet) and hence `configuredNetwork.nativeCurrency.symbol` also return `ETH` which is correct

Reply:
Ohh ohh, yes yes makes sense got confused with which balance we are showing tysm 🙌 resolving this then current situation makes more sense

Reply:
Yes I myself was confused initially 😅